frontend/public/components/overview/pods-overview.tsx

```diff
@@ -0,0 +1,78 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+import { ListGroup } from 'patternfly-react';
+
+import { ResourceLink, resourcePath, SidebarSectionHeading, StatusIconAndText } from '../utils';
+import {podPhase, PodKind, K8sResourceKind} from '../../module/k8s';
+
+const kind:string = 'Pod';
+const MAX_PODS:number = 3;
+
+const PodOverviewItem: React.FC<PodOverviewItemProps> = ({ pod }) => {
+  const { metadata: { name, namespace, uid } } = pod;
+  const phase = podPhase(pod);
+
+  return (
+    <li className="list-group-item container-fluid">
+      <div className="row">
+        <span className="col-xs-6">
+          <ResourceLink kind={kind} name={name} namespace={namespace} title={uid} />
+        </span>
+        <span className="col-xs-3">
+          <StatusIconAndText status={phase} />
+        </span>
+        <span className="col-xs-3">
+          <Link to={`${resourcePath(kind, name, namespace)}/logs`}>
+            View Logs
+          </Link>
+        </span>
+      </div>
+    </li>
+  );
+};
+
+PodOverviewItem.displayName = 'PodOverviewItem';
+
+type PodOverviewItemProps = {
+  pod: PodKind;
+};
+
+const PodsOverviewList: React.SFC<PodOverviewListProps> = ({ pods }) => (
+  <ListGroup componentClass="ul">
+    {_.map(pods, pod => <PodOverviewItem key={pod.metadata.uid} pod={pod} />)}
+  </ListGroup>
+);
+
+PodsOverviewList.displayName = 'PodsOverviewList';
+
+export const PodsOverview: React.SFC<PodsOverviewProps> = ({ pods, obj }) => {
+  const viewedPods:PodKind[] = _.slice(_.sortBy(pods, pod => _.get(pod, 'metadata.stats.startTime')), 0 , MAX_PODS) as PodKind[];
```

Comment:
The sort here looks wrong. `metadata.stats`? Do you want `status.startTime`? I wonder if we should use `creationTimestamp` though for pods that haven't started.

`_.take` might help here.

```suggestion
  const viewedPods: PodKind[] = _.take(_.orderBy(pods, pod => _.get(pod, ['metadata.creationTimestamp'], ['desc']), MAX_PODS);
```

Reply:
nice catch, should have been `metadata.status.startTime`
The design doc specifically requests pods with the most recent status

Reply:
I guess it depends on what is meant by status. Start time or any status update? Ordering by most recent container status update could cause the pods to jump around.

> should have been metadata.status.startTime

I think you want `status.startTime`. There's no `metadata.status` afaik.

Reply:
You'd probably want creation timestamp or name as a secondary sort so that the sort is stable. But I think it might be surprising if the pods reorder as they update.

Reply:
What we were going for is to show the "most recently updated status" for the pods as we thought that would be the most interesting. We don't want it to jump around, so perhaps we could either just do it for initial load or do some slow refresh (5-10 seconds) so it isn't jumping around too much.

We may want to evolve someday to show pods with error status then ordered by status.

Reply:
I don't think we can throttle updates for just this particular list.

Reply:
Result of out of band conversations:

Show all error pods or maximum of 3 pods when 3 or less error'd pods.
Sort by status, then by most recent container status, then by pod name.

Reply:
sounds like a good solution, thanks all